src/Data/Rational/Properties.agda

```diff
@@ -1719,6 +1720,91 @@ toℚᵘ-homo-∣-∣ (mkℚ -[1+ _ ] _ _) = *≡* refl
 ∣∣p∣∣≡∣p∣ : ∀ p → ∣ ∣ p ∣ ∣ ≡ ∣ p ∣
 ∣∣p∣∣≡∣p∣ p = 0≤p⇒∣p∣≡p (0≤∣p∣ p)
 
+------------------------------------------------------------------------
+-- Metric space
+------------------------------------------------------------------------
+
+private
+  d : ℚ → ℚ → ℚ
+  d p q = ∣ p - q ∣
+
+d-cong : Congruent _≡_ d
+d-cong = cong₂ _
+
+d-nonNegative : ∀ {p q} → 0ℚ ≤ d p q
+d-nonNegative {p} {q} = nonNegative⁻¹ _ {{∣-∣-nonNeg (p - q)}}
+
+d-definite : Definite _≡_ d
+d-definite {p} refl = cong ∣_∣ (+-inverseʳ p)
+
+d-indiscernable : Indiscernable _≡_ d
+d-indiscernable {p} {q} ∣p-q∣≡0 = begin
+  p               ≡˘⟨ +-identityʳ p ⟩
+  p - 0ℚ          ≡˘⟨ cong (_-_ p) (∣p∣≡0⇒p≡0 (p - q) ∣p-q∣≡0) ⟩
+  p - (p - q)     ≡⟨ cong (_+_ p) (neg-distrib-+ p (- q)) ⟩
```

Comment:
It feels like `p - (p - q) ≡ q` should be a lemma in `GroupProperties`?

Reply:
It actually requires commutativity! (this is hidden in `neg-distrib-+`). As far as I can tell it doesn't exist yet.

Reply:
The following proof looks commutativity-free to me?
```
p - (p - q) = p + - (p + - q) = p + (-p + q) = (p + -p) + q = 0# + q = q
```

Reply:
The step `- (p + - q) = - p + q` requires commutativity

Reply:
Oh whoops. Yes, you don't get distributivity for free.

Reply:
See `Algebra.Properties.AbelianGroup.⁻¹-anti-homo‿-` added in #2349 ?